docs/current_docs/manuals/developer/functions.mdx

```diff
@@ -1,21 +1,79 @@
 ---
-slug: /manuals/developer/python/753720/functions
+slug: /manuals/developer/functions
 displayed_sidebar: "current"
-toc_max_heading_level: 2
+toc_max_heading_level: 3
 title: "Functions"
 ---
-
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
 
 # Functions
 
-
 Dagger Functions are regular code, written in a supported programming language, and running in containers. Dagger Functions let you encapsulate common operations or workflows into discrete units with clear inputs and outputs.
 
 Here's an example of a simple Dagger Function:
 
-```python file=./snippets/functions/functions-simple/main.py
+<Tabs groupId="language">
+
+<TabItem value="Go">
+```go file=./snippets/functions/functions-simple/go/main.go
 ```
 
+:::note
+The context and error return are optional in the module's function signature; remove them if you don't need them.
+:::
```

Comment:
I expect folks to copy and paste these examples rather than build them up, so I propose keeping this here to introruce the concept early. I also think its worth introducing early since our init example comes with these inputs and return types too so its nice that that maps to this example.

Reply:
Sorry I don't agree to start with a function that has context.Context and returns an error without needing them and then rely on the note. Copy pasters that don't read the note will probably linger adding those when it's not needed, while if the first example doesn't have them but the second one does, even with just copy+paste (without the note) it's easier to understand.

Reply:
No problem, I'll move this around. Let me know what you think of the latest, it may just make sense to get rid of the note altogether.